doc/api/lines_api.rst

```diff
@@ -26,4 +26,4 @@ Functions
    :template: autosummary.rst
 
    segment_hits
-   
\ No newline at end of file
+   
```

Comment:
This seems to be adding a line with trailing whitespace, only to have it removed in a later remove-trailing-whitespace commit?

Reply:
@QuLogic I'm running the pre-commit hooks sequentially and applying their effects so that if someone did want to look at what was actually done (even though these will be (~mostly) ignored in `git blame`) they can see the effects instead of just getting a single commit that is just the result of applying all of the changes for

```console
$ pre-commit run --all-files
```

to pass. So sequentially that is `end-of-file-fixer` which enforces a newline followed by `trailing-whitespace` which removes the trailing whitespace that existed on the line that previously didn't have a newline following.

```console
$ git show c1a33a481b -- doc/api/lines_api.rst
commit c1a33a481b9c2df605bcb9bef9c19fe65c3dac21
Author: Matthew Feickert <matthew.feickert@cern.ch>
Date:   Fri Apr 8 16:02:24 2022 -0500

    STY: Apply end-of-file-fixer pre-commit hook

diff --git a/doc/api/lines_api.rst b/doc/api/lines_api.rst
index 808df726d1..44e0ec1cab 100644
--- a/doc/api/lines_api.rst
+++ b/doc/api/lines_api.rst
@@ -26,4 +26,4 @@ Functions
    :template: autosummary.rst

    segment_hits
-
\ No newline at end of file
+
$ git show 213061c080 -- doc/api/lines_api.rst
commit 213061c0804530d04bbbd5c259f10dc8504e5b2b
Author: Matthew Feickert <matthew.feickert@cern.ch>
Date:   Fri Apr 8 16:05:04 2022 -0500

    STY: Apply trailing-whitespace pre-commit hook

diff --git a/doc/api/lines_api.rst b/doc/api/lines_api.rst
index 44e0ec1cab..4cde67c7e6 100644
--- a/doc/api/lines_api.rst
+++ b/doc/api/lines_api.rst
@@ -26,4 +26,3 @@ Functions
    :template: autosummary.rst

    segment_hits
-
```

Their composite is then the last line in the file having the whitespace stripped and a newline added, seen here.

Is this what your question was on? Or did I misunderstand you?

Reply:
Yeah it unfortunately doesn't show up well in the diff. Thanks for clarifying! :)